--- origsrc/zstd-1.1.3/lib/Makefile	2016-09-27 17:15:03.000000000 -0500
+++ src/zstd-1.1.3/lib/Makefile	2016-10-05 16:03:30.048273700 -0500
@@ -50,6 +50,16 @@
 	SHARED_EXT_MAJOR = $(LIBVER_MAJOR).$(SHARED_EXT)
 	SHARED_EXT_VER = $(LIBVER).$(SHARED_EXT)
 	SONAME_FLAGS = -install_name $(LIBDIR)/libzstd.$(SHARED_EXT_MAJOR) -compatibility_version $(LIBVER_MAJOR) -current_version $(LIBVER)
+else ifeq ($(OS), Cygwin)
+	SHARED_PFX = cyg
+	SHARED_EXT = dll.a
+	SHARED_EXT_VER = -$(LIBVER_MAJOR).dll
+	SONAME_FLAGS = -Wl,--out-implib,libzstd.$(SHARED_EXT)
+else ifneq (,$(filter Windows%,$(OS)))
+	SHARED_PFX =
+	SHARED_EXT = dll.a
+	SHARED_EXT_VER = $(LIBVER_MAJOR).dll
+	SONAME_FLAGS = -Wl,--out-implib,libzstd.$(SHARED_EXT)
 else
 	SONAME_FLAGS = -Wl,-soname=libzstd.$(SHARED_EXT).$(LIBVER_MAJOR)
 	SHARED_EXT = so
@@ -77,9 +87,8 @@
 $(LIBZSTD): LDFLAGS += -shared -fPIC -fvisibility=hidden
 $(LIBZSTD): $(ZSTD_FILES)
 	@echo compiling dynamic library $(LIBVER)
-ifneq (,$(filter Windows%,$(OS)))
-	@$(CC) $(FLAGS) -DZSTD_DLL_EXPORT=1 -shared $^ -o dll\libzstd.dll
-	dlltool -D dll\libzstd.dll -d dll\libzstd.def -l dll\libzstd.lib
+ifneq (,$(filter Cygwin Windows%,$(OS)))
+	@$(CC) $(FLAGS) -shared $^ -DPIC $(SONAME_FLAGS) -o $(SHARED_PFX)zstd$(SHARED_EXT_VER)
 else
 	@$(CC) $(FLAGS) $^ $(LDFLAGS) $(SONAME_FLAGS) -o $@
 	@echo creating versioned links
@@ -130,6 +139,7 @@
 PREFIX      ?= $(prefix)
 exec_prefix ?= $(PREFIX)
 libdir      ?= $(exec_prefix)/lib
+DLLDIR      ?= $(PREFIX)/bin
 LIBDIR      ?= $(libdir)
 includedir  ?= $(PREFIX)/include
 INCLUDEDIR  ?= $(includedir)
@@ -146,9 +156,9 @@ else
 INSTALL ?= install
 endif
 
-INSTALL_PROGRAM ?= $(INSTALL)
+INSTALL_PROGRAM ?= $(INSTALL) -m 755
 INSTALL_DATA    ?= $(INSTALL) -m 644
-
+INSTALL_LIB     ?= $(INSTALL) -m 755
 
 libzstd.pc:
 libzstd.pc: libzstd.pc.in
@@ -164,9 +174,15 @@
 	@$(INSTALL_DATA) libzstd.pc $(DESTDIR)$(PKGCONFIGDIR)/
 	@echo Installing libraries
 	@$(INSTALL_DATA) libzstd.a $(DESTDIR)$(LIBDIR)
+ifneq (,$(filter Cygwin Windows%,$(OS)))
+	@$(INSTALL_LIB) -d $(DESTDIR)$(DLLDIR)/
+	$(INSTALL_LIB) $(SHARED_PFX)zstd$(SHARED_EXT_VER) $(DESTDIR)$(DLLDIR)/
+	$(INSTALL_DATA) libzstd.$(SHARED_EXT) $(DESTDIR)$(LIBDIR)/
+else
 	@$(INSTALL_PROGRAM) $(LIBZSTD) $(DESTDIR)$(LIBDIR)
 	@ln -sf $(LIBZSTD) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT_MAJOR)
 	@ln -sf $(LIBZSTD) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT)
+endif
 	@echo Installing includes
 	@$(INSTALL_DATA) zstd.h $(DESTDIR)$(INCLUDEDIR)
 	@$(INSTALL_DATA) common/zstd_errors.h $(DESTDIR)$(INCLUDEDIR)
@@ -177,8 +193,13 @@
 uninstall:
 	@$(RM) $(DESTDIR)$(LIBDIR)/libzstd.a
 	@$(RM) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT)
+ifneq (,$(filter Cygwin Windows%,$(OS)))
+	@$(RM) $(DESTDIR)$(DLLDIR)/$(SHARED_PFX)zstd$(SHARED_EXT_VER)
+	@$(RM) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT)
+else
 	@$(RM) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT_MAJOR)
 	@$(RM) $(DESTDIR)$(LIBDIR)/$(LIBZSTD)
+endif
 	@$(RM) $(DESTDIR)$(PKGCONFIGDIR)/libzstd.pc
 	@$(RM) $(DESTDIR)$(INCLUDEDIR)/zstd.h
 	@$(RM) $(DESTDIR)$(INCLUDEDIR)/zstd_errors.h

