diff -Naur zstd-1.1.0.orig/lib/Makefile zstd-1.1.0.new/lib/Makefile
--- zstd-1.1.3.orig/lib/Makefile	2016-09-27 22:15:03.000000000 +0000
+++ zstd-1.1.3.new/lib/Makefile	2016-10-06 10:56:25.612493000 +0000
@@ -159,7 +159,7 @@
              -e 's|@VERSION@|$(VERSION)|' \
              $< >$@
 
-install: libzstd.a libzstd libzstd.pc
+install: libzstd.pc
 	@$(INSTALL) -d -m 755 $(DESTDIR)$(PKGCONFIGDIR)/ $(DESTDIR)$(INCLUDEDIR)/
 	@$(INSTALL_DATA) libzstd.pc $(DESTDIR)$(PKGCONFIGDIR)/
 	@echo Installing libraries

